### PR TITLE
Barycentric supporting interpolation in different modes

### PR DESCRIPTION
**Pull request by AMD-dwang**
gl_Barycentric requires to be used as a parameter of interpolateAt*,

which means we use sample/centriod mode to update the IJCoordnate.

We don't clear if interpolateAtOffset can be used, so this change don't

include it.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit cb41ad5b252391ae38b3ceabbcc276e64751f36c
<details><summary>CTS tests (Failed: 0/179969)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35764/63747 (56.1%)</li>
<li>Failed: 0/63747 (0.0%)</li>
<li>Not Supported: 27983/63747 (43.9%)</li><li>Warnings: 0/63747 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27854/52475 (53.1%)</li>
<li>Failed: 0/52475 (0.0%)</li>
<li>Not Supported: 24621/52475 (46.9%)</li><li>Warnings: 0/52475 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38007/63747 (59.6%)</li>
<li>Failed: 0/63747 (0.0%)</li>
<li>Not Supported: 25740/63747 (40.4%)</li><li>Warnings: 0/63747 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchInOutImportExport.cpp [hunk 1]
```diff
@@ -2331,9 +2331,9 @@ Value *PatchInOutImportExport::patchGsBuiltInInputImport(Type *inputTy, unsigned
 //
 // @param inputTy : Type of input value
 // @param builtInId : ID of the built-in variable
-// @param sampleId : Sample ID; only needed for BuiltInSamplePosOffset
+// @param commonVal : Sample ID, only needed for BuiltInSamplePosOffset; InterpLoc, only needed for BuiltInBaryCoord
 // @param insertPos : Where to insert the patch instruction
-Value *PatchInOutImportExport::patchFsBuiltInInputImport(Type *inputTy, unsigned builtInId, Value *sampleId,
+Value *PatchInOutImportExport::patchFsBuiltInInputImport(Type *inputTy, unsigned builtInId, Value *commonVal,
```
Comment (linqun) [inline lgc/patch/PatchInOutImportExport.cpp:2336]: I prefer to use same name. i.e. commonVal -> generalVal

**Review by linqun — APPROVED** [review]
looks good to me for others.

**Comment by amdvlk-admin** [comment]
### Test summary for commit afec186189cba5b3cc0eed85ba79a450edae3961
<details><summary>CTS tests (Failed: 0/179658)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35420/63739 (55.6%)</li>
<li>Failed: 0/63739 (0.0%)</li>
<li>Not Supported: 28319/63739 (44.4%)</li><li>Warnings: 0/63739 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27611/52180 (52.9%)</li>
<li>Failed: 0/52180 (0.0%)</li>
<li>Not Supported: 24569/52180 (47.1%)</li><li>Warnings: 0/52180 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37681/63739 (59.1%)</li>
<li>Failed: 0/63739 (0.0%)</li>
<li>Not Supported: 26058/63739 (40.9%)</li><li>Warnings: 0/63739 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**